### DISCUSSION TITLE
Resource release problem in OpenTCS v5.16

### DISCUSSION THREAD

**Discussion by csliao0524** [openTCS/opentcs#145] [Q&A] [answered]
As shown in the attached image, when I try to dispatch different vehicles from different position to Point-0023, e.g.,

Vehicle-0001 (from Point-0007 to Point-0023)
Vehicle-0002 (from Point-0006 to Point-0023)

![1](https://github.com/openTCS/opentcs/assets/106263293/a24c0e26-f3ad-4f48-ae40-0e7929c1104e)
![2](https://github.com/openTCS/opentcs/assets/106263293/c295140e-2a3f-49d6-9710-245792f2f8e2)

Both cases hold the resources from Point-0007 to Point-0023 (occupying too many resources), which does not happen with other routes 
(e.g., dispatching Vehicle-0001 from Point-0023 to Point-0029 as shown in the image).

Has anyone encountered a similar situation? Is this behavior normal or is it a bug? Are there any solutions?

Thank you for your help!

**Comment by swltr** [comment]
@csliao0524 There was an issue related to resource management in 5.16 that has been fixed in 5.17. Can you still reproduce that situation with 5.17?

**Comment by csliao0524** [comment]
Hi @swltr 
Below are the test results using the 5.17, it seems like the vehicles still hold the unnecessary resources at the specific route. 
In addition, I've tried to assign the vehicle by ways of  "Sent to location" and "transportOrder web api",  both have same problem.
I'm not sure how to tackle with this issue, any advice?
Thanks for your reply!  

![image](https://github.com/openTCS/opentcs/assets/106263293/7a807933-7069-4488-b27d-0b392aafcb04)
![image](https://github.com/openTCS/opentcs/assets/106263293/58950f8a-1473-4fee-b90c-2751ff6a7f7a)
![image](https://github.com/openTCS/opentcs/assets/106263293/ffd5f29e-2571-4a6f-80e5-e2ed66014501)

**Reply by martingr** [reply, marked as answer]
@csliao0524 How long are the paths between Point-0007 and Point-0023? The rulers would suggest that each path has a length of 10.000 mm. Is this what's acutally configured in the "length" attribute of the paths?

**Comment by csliao0524** [comment]
Hi @martingr 
It seems the issue was related to the path length. I neglected to check and recalculate the path length when adjusting the layout, resulting in many paths being shorter than 10,000 mm.

![image](https://github.com/openTCS/opentcs/assets/106263293/e4598d5a-cfe9-4fbd-870b-25104c1d4a6e)
![image](https://github.com/openTCS/opentcs/assets/106263293/0c87d069-dcb8-4581-9388-219867e4d1df)
![image](https://github.com/openTCS/opentcs/assets/106263293/1cc97fd9-17f4-49d7-a954-0828f0a855a9)
![image](https://github.com/openTCS/opentcs/assets/106263293/74481a7a-98ad-4931-a3b3-8ebf1b296e4a)

After recalculating the path length, all paths returned to their normal length of 10000 mm. And the test result shows that the resources now can be released correctly for both versions 5.16 and 5.17(as shown in the attached image). 

![image](https://github.com/openTCS/opentcs/assets/106263293/97617092-8fac-417e-a3f4-34273939f71e)
![image](https://github.com/openTCS/opentcs/assets/106263293/d80b392d-8a1f-4148-af57-2ced4e9cdda1)
![image](https://github.com/openTCS/opentcs/assets/106263293/c576ebe5-4573-4d3c-9941-392479856777)

Additionally, could you please let me know where the suggestion of 10,000 mm from the ruler is documented in the official files? I don't seem to recall seeing it. 

Thank you for your assistance!

**Reply by martingr** [reply]
I'm glad I could help and that this fixed your issue.

As for your questing regarding the ruler: There is no specific documentation on this. I simply looked at the ruler in one of your screenshots (see the green rectangle):
![333059062-7a807933-7069-4488-b27d-0b392aafcb04](https://github.com/openTCS/opentcs/assets/24974244/d5ed0a99-086a-40f4-b207-b7e03cbebe73)

**Comment by csliao0524** [comment]
Ok! thanks for your reply, Although the issue is fixed, I still can not figure out why the incorrect length configuration would result in this resource problem? Any idea?

**Reply by martingr** [reply]
Since version 5.9, openTCS makes use of the vehicle's length for resources management. When releasing resources after a vehicle has completed a movement command, the vehicle's length is considered to decide which resources are actually not required any more.

In your example
![333059429-ffd5f29e-2571-4a6f-80e5-e2ed66014501](https://github.com/openTCS/opentcs/assets/24974244/2237f78a-3655-48d7-9092-484b1085c614)
the paths from Point-0008 to Point-0023 had a combined length of 3 mm, which was not enough to cover the vehicle's length of 1000 mm. In such cases, openTCS will keep paths behind the vehicle allocated until the length of the vehicle is fully coverered according the the length of the paths.
